plugins/arStorageServicePlugin/lib/arStorageServiceUtils.class.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class arStorageServiceUtils
+{
+  const STORAGE_SERVICE_PACKAGE_PATH = 'file';
+
+  /**
+   * Use phpcurl to request a URL and pass the header and stream back to
+   * the browser.
+   *
+   */
+  public static function getFileFromStorageService($url)
+  {
+    $ch = curl_init();
+
+    curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'GET');
+
+    curl_setopt($ch, CURLOPT_URL, $url);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 500);
+    curl_setopt($ch, CURLOPT_WRITEFUNCTION, function($ch, $data) {
+        echo $data;
+
+        return strlen($data);
+    });
+
+    curl_setopt($ch, CURLOPT_HEADERFUNCTION, function($ch, $header) {
+        header($header);
+
+        return strlen($header);
+    });
+
+    curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1); // Storage service redirects
+    curl_setopt($ch, CURLOPT_FAILONERROR, true);
+    curl_setopt($ch, CURLOPT_HTTPHEADER, array(
+      sprintf('Authorization: ApiKey %s:%s',
+      (string) QubitSetting::getByName('storage_service_username'),
+      (string)QubitSetting::getByName('storage_service_api_key')),
```

Comment:
Nitpick: needs a space after `(string)`

Reply:
fixed

Reply:
oh, one more style nitpick: I think it would be more readable to indent the sprintf arguments, and drop the closing parenthesis to the next line, e.g. 

```
    curl_setopt($ch, CURLOPT_HTTPHEADER, array(
      sprintf(
        'Authorization: ApiKey %s:%s',
        (string) QubitSetting::getByName('storage_service_username'),
        (string0) QubitSetting::getByName('storage_service_api_key')
      ),
      'User-Agent: DRMC',
    ));
```

Reply:
I agree - fixed.